src/explorer/explorerTreeController.ts

```diff
@@ -62,20 +73,29 @@ export default class ExplorerTreeController
       this._onTreeItemUpdate();
     });
 
-    treeView.onDidExpandElement(async (event: any): Promise<void> => {
-      log.info('Connection tree item was expanded', {
-        connectionId: event.element.connectionId,
-        connectionName: event.element.label,
-        isExpanded: event.element.isExpanded,
+    treeView.onDidExpandElement(async (event): Promise<void> => {
+      const treeItem = event.element as vscode.TreeItem &
+        TreeItemParentInterface;
+      this._telemetryService.track(
+        new TreeItemExpandedTelemetryEvent(treeItem)
+      );
+
+      log.info('Explorer tree item was expanded', {
+        type: treeItem.contextValue,
+        connectionName: treeItem.label,
+        isExpanded: treeItem.isExpanded,
       });
 
-      if (!event.element.onDidExpand) {
+      if (!treeItem.onDidExpand) {
```

Comment:
do I understand correctly that if we do not have an `onDidExpand` set then we also don't end up calling `_onTreeItemUpdate`? Seems to be the legacy behavior so not an issue for this PR, more of not sure why this is fully returning here.

Reply:
I think this part handles three elements that don't have children. I might be mistaken, maybe Rhys knows better.